src/components/views/rooms/RoomList.js

```diff
@@ -163,6 +157,11 @@ module.exports = React.createClass({
         this.setState(newState);
     },
 
+    onRoomReceipt: function(receiptEvent, room) {
+        // because if we read a notification, it will affect notification count
+        this.refreshRoomList();
```

Comment:
We should check if it is the user's receipt and not someone else then in order to prevent refreshing the room list for *every* receipt.